### PR TITLE
Clojure plugin docstrings

### PR DESCRIPTION
**Pull request by carocad**
Hey guys,
while researching a bit about watches, I found out a lot about the way LT and especially the Clojure plugins evaluates and watches source code. As suggesteed by @kenny-evitt I documented the related functions in hope for the future ;)

Hope it helps

### CODE REVIEW

**Comment by kenny-evitt** [comment]
@carocad Cool and thanks! I'll take a longer look soonish.

File: lein-light-nrepl/src/lighttable/nrepl/eval.clj [hunk 1]
```diff
@@ -70,7 +80,9 @@
      (and (string? res) (:verbatim opts)) res
      :else (pr-str res))))
 
-(defn truncate [v]
+(defn truncate
```
Comment (kenny-evitt) [inline lein-light-nrepl/src/lighttable/nrepl/eval.clj]: I can't find any references to this function. I think it's safe to remove, ignore for now, or just change the docstring to reflect that it's not used.

**Comment by kenny-evitt** [comment]
@carocad I'm going to defer to @cldwalker and @rundis about whether to merge this as they've been more active in maintaining this code.

Would you mind writing up some of what you learned, or already knew, about watches – in a separate (Markdown) document too? That's what I was expecting originally.

**Comment by carocad** [comment]
@kenny-evitt initially I thought the same: put all of it in a separate file. Nevertheless [as you suggested] (https://gitter.im/LightTable/LightTable?at=576e8e329717171554415492) and, after seeing the strong relation of watches to how LT works in general I thought that it would be better to actually document the source code.

I already added the relevant links to my sources on the PR. Those blogs and Google group post explain it clear and better that I could with a simple markdown document, so I don't see a reason to put it in a separate document now.

There is only left to document the `watched-range` function inside LT itself. I will do that one soon ...

**Comment by kenny-evitt** [comment]
@carocad Sorry for not being clear – the docstrings and comments are great but an overview document would be helpful too to understand how things fit together. I'm thinking of that document like an integration test versus the docstrings and comments like unit tests. An initial version could just quote, or even copy, the blog posts and discussion thread verbatim.

**Comment by sbauer322** [comment]
@carocad - I find the work done here quite useful and it is a pity this was never merged... 

Would you be interested in finishing this PR or would you prefer that I take it over?

**Comment by carocad** [comment]
@sbauer322 You can take over if you so wish to. I have made quite some PR that would definitely appreciate some love ;)

**Comment by sbauer322** [comment]
@cldwalker if you were planning on taking this over, it might be beneficial to use the `:doc` key I am adding for issue https://github.com/LightTable/LightTable/issues/2206... it was the reason I started on that task before this one! :smile: 

Since you assigned yourself to this task, do let me know if you were going to take it on.

**Comment by cldwalker** [comment]
@sbauer322 Good to know. I'll hold off until LightTable/LightTable#2206 is sorted out. I'm also happy to review if you want to take this on

**Comment by sbauer322** [comment]
Closing as #93 has been opened with some additional changes.

**Pull request closed without merging**